airbyte-integrations/connectors/destination-google-sheets/destination_google_sheets/buffer.py

```diff
@@ -0,0 +1,124 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+from typing import Any, List, Mapping
+
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.models import AirbyteStream
+
+
+class WriteBuffer:
+
+    # Default instance of AirbyteLogger
+    logger = AirbyteLogger()
+    # Buffer for input records
+    records_buffer = []
+    # Placeholder for streams metadata
+    stream_info = []
+    # interval after which the records_buffer should be cleaned up for selected stream
+    flush_interval = 1000
+
+    @property
+    def default_missing(self) -> str:
+        """
+        Default value for missing keys in record stream, compared to configured_stream catalog.
+        Overwrite if needed.
+        """
+        return ""
+
+    def buffer_stream(self, configured_stream: AirbyteStream):
+        """
+        Saves important stream's information for later use.
+
+        Particulary, creates the data structure for `records_stream`.
+        Populates `stream_info` placeholder with stream metadata information.
+        """
+        stream = configured_stream.stream
+        stream_schema = stream.json_schema
+        stream_name = stream.name
+
+        self.records_buffer.append({stream_name: []})
+        self.stream_info.append({stream_name: sorted(list(stream_schema.get("properties").keys())), "is_set": False})
+
+    def add_to_buffer(self, stream_name: str, record: Mapping):
+        """
+        Populates input records to `records_buffer`.
+
+        1) normalizes input record
+        2) coerces normalized record to str
+        3) gets values as list of record values from record mapping.
+        """
+        for stream in self.records_buffer:
+            if stream_name in stream:
```

Comment:
btw, it is possbile that stream_name can be missing in records_buffer?

Reply:
We are working with configured_catalog, so - no.